Yank/multistate/multistateanalyzer.py

```diff
@@ -692,7 +729,7 @@ def _update_cache(self, key, new_value, check_changes=False):
         except KeyError:
             invalidate_cache = False
         else:
-            if check_changes and old_value == new_value:
+            if check_changes and self._check_equal(old_value, new_value):
```

Comment:
Just for curiosity, what does this do differently than `==` equality?

Reply:
Numpy arrays dont do `if array == array` without a `any` or `all`, and `all` does not work on non iterable.